src/pocketmine/block/Chest.php

```diff
@@ -76,7 +77,9 @@ public function place(Item $item, Block $blockReplace, Block $blockClicked, int
 					$c = $this->getSide($side);
 					if($c instanceof Chest and $c->isSameType($this) and $c->facing === $this->facing){
 						$pair = $this->level->getTile($c);
-						if($pair instanceof TileChest and !$pair->isPaired()){
+						$ev = new ChestPairEvent($this, $c);
+                        $ev->call();
+						if($pair instanceof TileChest and !$ev->isCancelled() and !$pair->isPaired()){
```

Comment:
This logic order makes no sense. A plugin will receive this event when a pairing is not going to happen due to the target already being paired.